vm/systemSmartContracts/auction_test.go

```diff
@@ -0,0 +1,510 @@
+package systemSmartContracts
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"math/rand"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/vm"
+	"github.com/ElrondNetwork/elrond-go/vm/mock"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+	"github.com/stretchr/testify/assert"
+)
+
+func createABid(totalStakeValue uint64, numBlsKeys uint32, maxStakePerNode uint64) AuctionData {
+	data := AuctionData{
+		RewardAddress:   []byte("addr"),
+		StartNonce:      0,
+		Epoch:           0,
+		BlsPubKeys:      nil,
+		TotalStakeValue: big.NewInt(0).SetUint64(totalStakeValue),
+		BlockedStake:    big.NewInt(100000000),
+		MaxStakePerNode: big.NewInt(0).SetUint64(maxStakePerNode),
+	}
+
+	keys := make([][]byte, 0)
+	for i := uint32(0); i < numBlsKeys; i++ {
+		keys = append(keys, []byte(fmt.Sprintf("%d", rand.Uint32())))
+	}
+	data.BlsPubKeys = keys
+
+	return data
+}
+
+func TestAuctionSC_calculateNodePrice_Case1(t *testing.T) {
```

Comment:
What is this Case1? Should add comment describing what you are testing. Explain all other cases found in this file

Reply:
the cases can be determined from the input bids.